src/FoundationClasses/TKernel/OSD/OSD_PerfMeter.cxx

```diff
@@ -15,389 +15,366 @@
-Function :      perf_init_meter
-Purpose  :      Creates new counter (if it is absent) identified by
-                MeterName and resets its cumulative value
-Returns  :      iMeter if OK, -1 if alloc problem
-======================================================================*/
-int perf_init_meter(const char* const MeterName)
+public:
+  // Constructor initializes the stopwatch.
+  // It does not start the stopwatch.
+  Stopwatch();
+
+  // Starts the stopwatch. If it is already running, it resets the start time.
+  void Start();
+
+  // Stops the stopwatch. If it is already stopped, it does nothing.
+  void Stop();
+
+  // Returns the elapsed time in milliseconds since the stopwatch was started.
+  // If the stopwatch is still running, it returns the time since the last start.
+  // If the stopwatch was stopped, it returns the time between the last start and stop.
+  // If the stopwatch was never started, it returns 0.
+  std::chrono::milliseconds Elapsed() const;
+
+  // Returns true if the stopwatch is currently running.
+  // Returns false if the stopwatch is stopped or was never started.
+  inline bool IsRunning() const;
+
+  // Returns true if the stopwatch has been started at least once and has a non-zero elapsed time.
+  // Returns false if the stopwatch was never started or has zero elapsed time.
+  inline bool IsActive() const;
+
+private:
+  std::chrono::steady_clock::time_point myStartTime; //< The time when the stopwatch was started.
+  std::chrono::steady_clock::time_point myEndTime;   //< The time when the stopwatch was stopped.
+                                                     //  Equal to myStartTime if the stopwatch is
+                                                     //  running or was never started.
+  bool myIsTicking;                                  //< Indicates whether the stopwatch is
+                                                     //  currently running.
+};
+
+//==================================================================================================
+
+Stopwatch::Stopwatch()
+    : myStartTime(std::chrono::steady_clock::now()),
+      myEndTime(myStartTime),
+      myIsTicking(false)
 {
-  return _perf_init_meter(MeterName, ~0);
 }
 
-/*======================================================================
-Function :      perf_tick_meter
-Purpose  :      Increments the counter of meter MeterName without changing
-                its state with respect to measurement of time.
-        creates new meter if there is no such meter
-Returns  :      iMeter if OK, -1 if no such meter and cannot create a new one
-======================================================================*/
-int perf_tick_meter(const char* const MeterName)
+//==================================================================================================
+
+void Stopwatch::Start()
 {
-  int ic = find_meter(MeterName);
+  myStartTime = std::chrono::steady_clock::now();
```

Comment:
Migration from OSD_Chronometer to chrono is totally different at this realization.
Need to return to OSD_Chronometer.
The difference is a way of calculation of the time (based on CPU usage of single thread or special timer).